janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/StandardJanusGraphTx.java

```diff
@@ -1588,7 +1589,13 @@ public void expireSchemaElement(final long id) {
         if (vertexCache.contains(id)) {
             final InternalVertex v = vertexCache.get(id, externalVertexRetriever);
             if (v instanceof JanusGraphSchemaVertex) {
-                ((JanusGraphSchemaVertex) v).resetCache();
+                JanusGraphSchemaVertex sv = (JanusGraphSchemaVertex) v;
+                sv.resetCache();
+                if (sv.getDefinition().containsKey(TypeDefinitionCategory.INTERNAL_INDEX)) {
+                    String store = sv.getDefinition().getValue(TypeDefinitionCategory.BACKING_INDEX,String.class);
+                    IndexTransaction idxTx = txHandle.getIndexTransaction(store);
+                    idxTx.invalidate(sv.name());
```

Comment:
Thanks, I believe this works but not ideal. If I understand it correctly, it clears cache for the whole mixed index store, which includes caches for many other indices.